gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -232,20 +230,19 @@ public void run() {
 
       try {
         if (shouldPublishDataInTask()) {
-
           // If data should be published by the task, publish the data and set the task state to COMMITTED.
           // Task data can only be published after all forks have been closed by closer.close().
           publishTaskData();
           this.taskState.setWorkingState(WorkUnitState.WorkingState.COMMITTED);
```

Comment:
If the job commit policy is `COMMIT_ON_PARTIAL_SUCCESS`, we will still commit output data of failed tasks, so setting task state to `COMMITTED` does not make nonsense. Also the client/driver will only check individual task states in `JobContext` if and only if the commit policy is `COMMIT_ON_FULL_SUCCESS`. 

I think we need a meeting to go through the different commit policies and their relationship with the final task/job states and make the semantics clear and stick to such semantics. This is currently a quite error-prone and not intuitive at all.